testApp/pv/testIntrospect.cpp

```diff
@@ -146,6 +146,33 @@ static void testStructure()
     testOk1(struct1->getFieldName(0)==names1[0]);
     testOk1(struct1->getFieldName(1)==names1[1]);
 
+    testOk1(struct1->getField("nonexistent").get()==NULL);
+    try {
+        FieldConstPtr field(struct1->getField(9999).get());
+        testFail("missing expected exception");
+    } catch (std::out_of_range& e) {
+        testPass("caught expected exception: %s", e.what());
+    }
+
+    testOk1(struct1->getFieldT("innerA")==fields1[0]);
+    testOk1(struct1->getFieldT("innerB")==fields1[1]);
+    testOk1(struct1->getFieldT(0)==fields1[0]);
+    testOk1(struct1->getFieldT(1)==fields1[1]);
+
+    try {
+        FieldConstPtr field(struct1->getFieldT("nonexistent"));
+        testFail("missing required exception");
+    } catch (std::runtime_error& e) {
+        testPass("caught expected exception: %s", e.what());
+    }
+
+    try {
+        FieldConstPtr field(struct1->getFieldT(9999));
+        testFail("missing required exception");
+    } catch (std::runtime_error& e) {
+        testPass("caught expected exception: %s", e.what());
+    }
+
```

Comment:
Should also test the casting variants in the case where a member exists, but not with the requested type.